patches/node/build_add_gn_build_files.patch

```diff
@@ -449,6 +449,255 @@ index 0000000000000000000000000000000000000000..a47875642d8f825c84ba1e82e3892a97
 +    ":tar_headers",
 +  ]
 +}
+diff --git a/deps/base64/BUILD.gn b/deps/base64/BUILD.gn
+new file mode 100644
+index 0000000000000000000000000000000000000000..0796c0b7d85f89ed01c392b2ee971bcf5e96b94c
+--- /dev/null
++++ b/deps/base64/BUILD.gn
+@@ -0,0 +1,215 @@
++config("base64_config") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib",
++  ]
++
++  defines = [ "BASE64_STATIC_DEFINE" ]
++}
++
++static_library("base64") {
++  defines = [ "BASE64_STATIC_DEFINE" ]
++  deps = []
++
++  public_configs = [ ":base64_config" ]
++
++  include_dirs = [
++    "base64/include",
++    "base64/lib",
++  ]
++
++  cflags_c = [
++    "-Wno-implicit-fallthrough",
++    "-Wno-unused-but-set-variable",
++    "-Wno-shadow",
++  ]
++
++  sources = [
++    "base64/include/libbase64.h",
++    "base64/lib/arch/generic/codec.c",
++    "base64/lib/tables/tables.c",
++    "base64/lib/codec_choose.c",
++    "base64/lib/codecs.h",
++    "base64/lib/lib.c",
++  ]
++
++  if (target_cpu == "arm") {
++    defines += ["HAVE_NEON32=1"]
++    deps += [":base64_neon32"]
++  } else {
++    sources += [ "base64/lib/arch/neon32/neon32_codec.c" ]
++  }
++
++  if (target_cpu == "arm64") {
++    defines += ["HAVE_NEON64=1"]
++    deps += [":base64_neon64"]
++  } else {
++    sources += [ "base64/lib/arch/neon64/neon64_codec.c" ]
++  }
++
++  if (target_cpu == "ia32" || target_cpu == "x64" || target_cpu == "x32") {
++    defines += [
++      "HAVE_SSSE3=1",
++      "HAVE_SSE41=1",
++      "HAVE_SSE42=1",
++      "HAVE_AVX=1",
++      "HAVE_AVX2=1",
++    ]
++
++    deps += [
++      ":base64_ssse3",
++      ":base64_sse41",
++      ":base64_sse42",
++      ":base64_avx",
++      ":base64_avx2",
++    ]
++  } else {
++    sources += [
++      "base64/lib/arch/ssse3/ssse3_codec.c",
++      "base64/lib/arch/sse41/sse41_codec.c",
++      "base64/lib/arch/sse42/sse42_codec.c",
++      "base64/lib/arch/avx/avx_codec.c",
++      "base64/lib/arch/avx2/avx2_codec.c",
++    ]
++  }
++}
++
++static_library("base64_ssse3") {
++  include_dirs = [
++    "base64/include",
++    "base64/lib"
++  ]
++
++  defines = [
++    "BASE64_STATIC_DEFINE",
++    "HAVE_SSSE3=1"
++  ]
++
++  cflags_c = [ "-Wno-implicit-fallthrough" ]
++
++  sources = [ "base64/lib/arch/ssse3/ssse3_codec.c" ]
++
++  if (!is_win || is_clang) {
```

Comment:
These are not exclusive in Electron, clang is enabled across all OS we support. Should this just be `is_clang` ?